frontend/src/metabase/query_builder/components/view/NativeVariablesButton/NativeVariablesButton.tsx

```diff
@@ -1,24 +1,28 @@
-/* eslint-disable react/prop-types */
 import { t } from "ttag";
 import { Icon } from "metabase/core/components/Icon";
 import Tooltip from "metabase/core/components/Tooltip";
-import NativeQuery from "metabase-lib/queries/NativeQuery";
+import type Question from "metabase-lib/Question";
 import { ButtonRoot } from "./NativeVariablesButton.styled";
 
+interface NativeVariablesButtonProps {
+  className?: string;
+  isShowingTemplateTagsEditor: boolean;
+  size: number;
+  toggleTemplateTagsEditor: () => void;
+}
+
 export const NativeVariablesButton = ({
-  toggleTemplateTagsEditor,
-  isShowingTemplateTagsEditor,
   className,
+  isShowingTemplateTagsEditor,
   size,
-}) => (
+  toggleTemplateTagsEditor,
+}: NativeVariablesButtonProps) => (
   <Tooltip tooltip={t`Variables`}>
     <ButtonRoot className={className} isSelected={isShowingTemplateTagsEditor}>
       <Icon name="variable" size={size} onClick={toggleTemplateTagsEditor} />
     </ButtonRoot>
   </Tooltip>
 );
 
-NativeVariablesButton.shouldRender = ({ question }) =>
-  question.query() instanceof NativeQuery &&
-  question.database() &&
-  question.database().hasFeature("native-parameters");
+NativeVariablesButton.shouldRender = ({ question }: { question: Question }) =>
```

Comment:
Not sure if I missed something. But seems like `NativeVariablesButton.shouldRender` isn't used, so we should delete it instead in that case.

Reply:
Good call, I was trying to avoid any changes apart from typing, but that's a good one 👍

Reply:
OTOH, it seems to me we should be actually checking for the `native-parameters` feature when rendering the button 🤔 I'll remove it anyway because it's not used and I don't think we should be using this `shouldRender` approach, but I'll double check if the check itself makes sense